### PR TITLE
Switch pkg/image to use clientsets

### PR DESCRIPTION
**Pull request by soltysh**
@mfojtik @deads2k I hope this handles all of them, including the SAR bits we've talked about yesterday.

### CODE REVIEW

File: pkg/image/registry/imagestreamimport/rest.go [hunk 1]
```diff
@@ -104,33 +106,37 @@ func (r *REST) Create(ctx apirequest.Context, obj runtime.Object, _ bool) (runti
...
-	isCreateImage, err := r.sarClient.Create(authorizationapi.AddUserToSAR(user,
-		&authorizationapi.SubjectAccessReview{
-			Action: authorizationapi.Action{
-				Verb:     "create",
-				Group:    imageapi.GroupName,
-				Resource: "images",
+	createImageSAR := authorizationutil.AddUserToSAR(user, &authorizationapi.SubjectAccessReview{
+		Spec: authorizationapi.SubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationapi.ResourceAttributes{
+				Namespace: inputMeta.Namespace,
```
Comment (deads2k) [inline pkg/image/registry/imagestreamimport/rest.go]: Was this namespace scoped before?  `images` are cluster scoped.

File: pkg/image/registry/imagestreamimport/rest.go [hunk 2]
```diff
@@ -104,33 +106,37 @@ func (r *REST) Create(ctx apirequest.Context, obj runtime.Object, _ bool) (runti
...
-	isCreateImageStreamMapping, err := r.sarClient.Create(authorizationapi.AddUserToSAR(user,
-		&authorizationapi.SubjectAccessReview{
-			Action: authorizationapi.Action{
-				Verb:     "create",
-				Group:    imageapi.GroupName,
-				Resource: "imagestreammapping",
+	createImageStreamMappingSAR := authorizationutil.AddUserToSAR(user, &authorizationapi.SubjectAccessReview{
+		Spec: authorizationapi.SubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationapi.ResourceAttributes{
+				Namespace: inputMeta.Namespace,
```
Comment (deads2k) [inline pkg/image/registry/imagestreamimport/rest.go]: I expected this one to be namespace scoped, but it looks like it wasn't namespaced before.  Was it?

File: pkg/image/apis/image/helper.go [hunk 3]
```diff
@@ -355,6 +350,11 @@ func JoinImageStreamTag(name, tag string) string {
 	return fmt.Sprintf("%s:%s", name, tag)
 }
 
+// JoinImageStreamImage creates a name for image stream image object from an image stream name and an id.
+func JoinImageStreamImage(name, id string) string {
```
Comment (deads2k) [inline pkg/image/apis/image/helper.go]: Why the change?  `Make` reads better to me for this purpose.

File: pkg/image/prune/prune.go [hunk 4]
```diff
@@ -900,32 +900,32 @@ func pruneManifests(
...
+func NewImageDeleter(images imageclient.ImagesGetter) ImageDeleter {
 	return &imageDeleter{
 		images: images,
 	}
 }
 
 func (p *imageDeleter) DeleteImage(image *imageapi.Image) error {
 	glog.V(4).Infof("Deleting image %q", image.Name)
-	return p.images.Delete(image.Name)
+	return p.images.Images().Delete(image.Name, metav1.NewDeleteOptions(0))
```
Comment (deads2k) [inline pkg/image/prune/prune.go]: old code was this aggresive with the delete options?

**Comment by deads2k** [comment]
errors look real.

**Comment by deads2k** [comment]
I think the questions are still outstanding.

File: pkg/image/registry/imagestreamimport/rest.go (hunk 1 above)
Comment (soltysh) [reply pkg/image/registry/imagestreamimport/rest.go]: This is imagestreamimport, not image. Iow. you create a resource that tells the server to import some data inside particular image stream.

File: pkg/image/prune/prune.go (hunk 4 above)
Comment (soltysh) [reply pkg/image/prune/prune.go]: Hmmm... there was no such option in the old code, and since this is pruning I assumed this will be the best option.

File: pkg/image/apis/image/helper.go (hunk 3 above)
Comment (soltysh) [reply pkg/image/apis/image/helper.go]: I wanted to make it consistent with the [other helper](https://github.com/openshift/origin/blob/381a1436e1988e9419ee58d4282d1e84d479ec34/pkg/image/apis/image/helper.go#L351) `JoinImageStreamTag`.

File: pkg/image/registry/imagestreamimport/rest.go (hunk 1 above)
Comment (soltysh) [reply pkg/image/registry/imagestreamimport/rest.go]: I take that back. Yes, you're right this should not be namespaced.

File: pkg/image/registry/imagestreamimport/rest.go (hunk 2 above)
Comment (soltysh) [reply pkg/image/registry/imagestreamimport/rest.go]: It was not, I'm removing the namespace parameter.

File: pkg/image/apis/image/helper.go (hunk 3 above)
Comment (soltysh) [reply pkg/image/apis/image/helper.go]: I'll split it to a separate commit.

**Comment by soltysh** [comment]
@deads2k comments addressed, problems solved and rebase. ptal

File: pkg/image/prune/prune.go (hunk 4 above)
Comment (deads2k) [reply pkg/image/prune/prune.go]: > Hmmm... there was no such option in the old code, and since this is pruning I assumed this will be the best option.

pass nil or empty object.  Pretty sure this one takes a nil.

File: pkg/image/apiserver/apiserver.go [hunk 5]
```diff
@@ -124,19 +124,15 @@ func (c *ImageAPIServerConfig) newV1RESTStorage() (map[string]rest.Storage, erro
...
-	if err != nil {
-		return nil, err
-	}
-	deprecatedOpenshiftClient, err := osclient.New(c.CoreAPIServerClientConfig)
+	coreClient, err := coreclient.NewForConfig(c.CoreAPIServerClientConfig)
 	if err != nil {
 		return nil, err
 	}
-	coreClient, err := coreclient.NewForConfig(c.CoreAPIServerClientConfig)
+	authorizationClient, err := authorizationclient.NewForConfig(c.GenericConfig.LoopbackClientConfig)
```
Comment (deads2k) [inline pkg/image/apiserver/apiserver.go]: preexisting, but this one should probably be `CoreAPIServerClientConfig`

**Comment by deads2k** [comment]
The delete options and the client config bit.  Otherwise, lgtm.  After that feel free to tag it.

**Comment by soltysh** [comment]
Rebased, re-applying the label back.
